src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,68 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (isValid(user) && isLoginValid(user)
+                          && isPasswordValid(user)
+                          && isAgeValid(user)
+                          && isLoginInStorage(user)) {
+            storageDao.add(user);
+        }
         return null;
     }
+
+    private boolean isValid(User user) {
+        if (user == null) {
+            throw new NullPointerException("You didn't insert any data!");
```

Comment:
![image](https://github.com/mate-academy/jv-registration-validation/assets/63095184/bbdf28e9-79c4-4a76-a060-2642b9cdf659)

Reply:
For all cases